native/src/main/java/io/ballerina/lib/aws/mpe/AwsMpeClient.java

```diff
@@ -0,0 +1,130 @@
+ * utility methods to invoke as inter-op functions.
+ */
+public final class AwsMpeClient {
+    private static final String NATIVE_CLIENT = "nativeClient";
+    private static final ExecutorService EXECUTOR_SERVICE = Executors.newCachedThreadPool(new AwsMpeThreadFactory());
+
+    private AwsMpeClient() {
+    }
+
+    /**
+     * Creates an AWS MPE native client with the provided configurations.
+     *
+     * @param bAwsMpeClient The Ballerina AWS MPE client object.
+     * @param configurations AWS MPE client connection configurations.
+     * @return A Ballerina `mpe:Error` if failed to initialize the native client with the provided configurations.
+     */
+    public static Object init(BObject bAwsMpeClient, BMap<BString, Object> configurations) {
+        try {
+            ConnectionConfig connectionConfig = new ConnectionConfig(configurations);
+            AwsCredentials credentials = getCredentials(connectionConfig);
+            AwsCredentialsProvider credentialsProvider = StaticCredentialsProvider.create(credentials);
+            MarketplaceEntitlementClient nativeClient = MarketplaceEntitlementClient.builder()
+                    .credentialsProvider(credentialsProvider)
+                    .region(connectionConfig.region()).build();
+            bAwsMpeClient.addNativeData(NATIVE_CLIENT, nativeClient);
+        } catch (Exception e) {
+            String errorMsg = String.format("Error occurred while initializing the marketplace entitlement client: %s",
+                    e.getMessage());
+            return CommonUtils.createError(errorMsg, e);
+        }
+        return null;
+    }
+
+    private static AwsCredentials getCredentials(ConnectionConfig connectionConfig) {
+        if (Objects.nonNull(connectionConfig.sessionToken())) {
+            return AwsSessionCredentials.create(connectionConfig.accessKeyId(), connectionConfig.secretAccessKey(),
+                    connectionConfig.sessionToken());
+        } else {
+            return AwsBasicCredentials.create(connectionConfig.accessKeyId(), connectionConfig.secretAccessKey());
+        }
+    }
+
+    /**
+     * Retrieves the entitlement values for a given product.
+     *
+     * @param env The Ballerina runtime environment.
+     * @param bAwsMpeClient The Ballerina AWS MPE client object.
+     * @param request The Ballerina AWS MPE `GetEntitlement` request.
+     * @return A Ballerina `mpe:Error` if there was an error while processing the request or else the AWS MPE response.
+     */
+    public static Object getEntitlements(Environment env, BObject bAwsMpeClient, BMap<BString, Object> request) {
+        MarketplaceEntitlementClient nativeClient = (MarketplaceEntitlementClient) bAwsMpeClient
+                .getNativeData(NATIVE_CLIENT);
+        GetEntitlementsRequest entitlementsRequest = CommonUtils.getNativeRequest(request);
+        Future future = env.markAsync();
+        EXECUTOR_SERVICE.execute(() -> {
```

Comment:
Is there a reason to execute this logic in a separate thread? Can't we directly return the results without `markAsync()`?

Reply:
All network actions should be handled separately by using a different thread pool. Hence, we follow this approach.